pandas/core/generic.py

```diff
@@ -2530,6 +2535,22 @@ def to_json(
             date_format = "iso"
         elif date_format is None:
             date_format = "epoch"
+            dtypes = getattr(self.dtypes, "values", [self.dtypes])
+            if "m8[ns]" in dtypes or "M8[ns]" in dtypes:
```

Comment:
@jbrockmendel is this the best way to test if a dataframe contains any datetime dtypes?

Reply:
this looks fragile.  id suggest `any(lib.is_np_dtype(dtype, "mM") for dtype in dtypes)`.  also instead of using a getattr id suggest a more explicit `dtypes = self.dtypes.values if self.ndim == 2 else np.array([self.dtype], dtype=object)`